src/Libraries/CoreNodeModels/Range.cs

```diff
@@ -24,7 +24,14 @@ public class Range : NodeModel
         [JsonConstructor]
         private Range(IEnumerable<PortModel> inPorts, IEnumerable<PortModel> outPorts) : base(inPorts, outPorts)
         {
+            if (inPorts.Count() == 3)
```

Comment:
any reason for checking this? or is there another Range node that has < 3 inports?

Reply:
Unit test for this ?

Reply:
Checking for this and bail out if user modified graph causing inPorts does not match default numbers. This should be covered once we enable tests to run on JSON DYNs. We already have unit test for Range Node

Reply:
Sorry, but why user will modify the graph? I see that  range and Sequence has three ports and no provision it to remove from UI.

Reply:
User could do whatever, I simply do not want this code to crash when the graph somehow does not contain three ports..

Reply:
in that case, user can update the graph however they want..they can even add a new port...the error check is ok, but I don't see it very necessary considering that Range / sequence always have 3 ports.

Reply:
@ramramps Because we use `inPorts.ElementAt(2)` below, so it will crash if there are less than three ports. I can change the check to larger than 3 if you want

Reply:
I would think that the best thing to do in the JSON constructor is to add any missing input and output ports, in case the Graph was written by hand and is missing ports.

Reply:
It is there in the other constructor 
```
public Range()
        {
            InPorts.Add(new PortModel(PortType.Input, this, new PortData("start", Resources.RangePortDataStartToolTip, startPortDefaultValue)));
            InPorts.Add(new PortModel(PortType.Input, this, new PortData("end", Resources.RangePortDataEndToolTip, endPortDefaultValue)));
            InPorts.Add(new PortModel(PortType.Input, this, new PortData("step", Resources.RangePortDataStepToolTip, stepPortDefaultValue)));
            OutPorts.Add(new PortModel(PortType.Output, this, new PortData("seq", Resources.RangePortDataSeqToolTip)));

            RegisterAllPorts();

            ArgumentLacing = LacingStrategy.Auto;
        }
```

Reply:
@ramramps @gregmarr I'm setting InPorts and OutPorts when number of ports in Json does not match, let me know if you have different opinions

Reply:
LGTM